clients/web/src/server.js

```diff
@@ -10,22 +10,28 @@ import { restRequest } from 'girder/rest';
  */
 function restartServer() {
     function waitForServer() {
-        restRequest({
-            type: 'GET',
-            path: 'system/version',
-            error: null
-        }).done(_.bind(function (resp) {
-            if (resp.serverStartDate !== restartServer._lastStartDate) {
-                restartServer._reloadWindow();
-            } else {
-                window.setTimeout(waitForServer, 1000);
+        return new Promise((resolve, reject) => {
+            function wait() {
+                restRequest({
+                    type: 'GET',
+                    path: 'system/version',
+                    error: null
+                }).done(_.bind(function (resp) {
+                    if (resp.serverStartDate !== restartServer._lastStartDate) {
+                        resolve();
+                        restartServer._reloadWindow();
+                    } else {
+                        window.setTimeout(wait, 1000);
+                    }
+                })).error(_.bind(function () {
+                    window.setTimeout(wait, 1000);
+                }));
```

Comment:
I realize that you didn't write this code, but the `_.bind` here and a few lines above don't actually bind a context.  It would be better to just change these to arrow functions.